enterprise/frontend/src/embedding-sdk/components/public/dashboard/SdkDashboard.tsx

```diff
@@ -164,6 +172,14 @@ const SdkDashboardInner = ({
     ? "question"
     : renderModeState;
 
+  // Now only used when rerendering the dashboard after creating a new question from the dashboard.
+  const refetchDashboardRef = useRef(_.noop);
```

Comment:
Another way to deal with this is to use `key` to rerender the whole component, I think that might be a hack, and making the provider works without requiring unmounting is more ideal. Since we keep the rest of the state, and don't have to do unnecessary works if any.